pkg/analysis/passes/sdkusage/sdkusage.go

```diff
@@ -101,5 +125,80 @@ func run(pass *analysis.Pass) (interface{}, error) {
 		return nil, nil
 	}
 
+	if pluginGoSdkVersion == "" {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			goModError,
+			"go.mod can not be parsed from your source code",
+			"Your go.mod can not be parsed. Please make sure it is valid You can use `go mod tidy` to fix it.",
+		)
+		return nil, nil
+	}
+
+	latestRelease, err := githubutils.FetchLatestGrafanaSdkRelease()
+	if err != nil {
+		// it is most likely this failed because of github auth or rate limits
+		logme.Debugln(err)
+		return nil, nil
+	}
+
+	if latestRelease.TagName == pluginGoSdkVersion {
+		// plugin is using the latest version no further checks
+		return nil, nil
+	}
+
+	pluginGoSdkRelease, err := githubutils.FetchGrafanaSdkReleaseByTag(pluginGoSdkVersion)
+	if err != nil {
+		// it is most likely this failed because of github auth or rate limits
+		logme.Debugln(err)
+		return nil, nil
+	}
+
+	daysDiff, err := daysDifference(pluginGoSdkRelease.PublishedAt, latestRelease.PublishedAt)
```

Comment:
I think it's good to use as origing the release date of the latest release but maybe we should specify that somewhere? even if it's just a comment here.

This has a caveat though, if we don't release a version for a while (> 2 months) and a plugin is using an older version (e.g 3 months), it won't receive a warning. If the next day we release a version, it will start failing (which can be confusing if the plugin is in the review process). I wonder if it's better to just use the current day for the comparison here.

Reply:
Good idea. I'll change the logic to use the current date as the comparison date.